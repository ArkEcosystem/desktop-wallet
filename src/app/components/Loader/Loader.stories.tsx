import React from "react";

import { Loader } from "./Loader";

export default { title: "App / Components / Loader" };

export const Default = () => (
	<div className="ml-2 space-y-5">
		<div className="my-4">Absolute position in relative container div</div>
		<div className="relative border-1 border-theme-neutral-500" style={{ width: 600, height: 400 }}>
			<Loader />
		</div>
	</div>
);
