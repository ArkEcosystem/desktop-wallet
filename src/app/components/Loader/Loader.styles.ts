import tw, { styled } from "twin.macro";

export const LogoSpinner = styled.div`
	& {
		${tw`relative z-30`}
		width: 7rem;
		height: 7rem;

		.centered {
			${tw`flex items-center justify-center absolute top-0 right-0 left-0 bottom-0`}
		}

		.spin {
			animation: spin 0.8s ease-out infinite;
			&.left {
				animation: spin-left 1s ease-in infinite;
			}
		}

		@keyframes spin {
			from {
				transform: rotate(0deg);
			}
			to {
				transform: rotate(360deg);
			}
		}

		@keyframes spin-left {
			from {
				transform: rotate(0deg);
			}
			to {
				transform: rotate(-360deg);
			}
		}
	}
`;

export const LoaderWrapper = styled.div`
	${tw`absolute flex items-center justify-center top-0 right-0 left-0 bottom-0`}
`;
