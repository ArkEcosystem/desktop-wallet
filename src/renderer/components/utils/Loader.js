import { PulseLoader } from 'vue-spinner/dist/vue-spinner.min'

export default {
  functional: true,

  props: {
    color: {
      type: String,
      required: false
    }
  },

  render: (h, ctx) => h(PulseLoader, {
    props: {
      color: ctx.props.color
    }
  })
}
