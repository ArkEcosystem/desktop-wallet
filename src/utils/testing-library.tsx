import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment } from "@arkecosystem/platform-sdk-profiles";
import { render } from "@testing-library/react";
import { EnvironmentProvider } from "app/contexts";
import { i18n } from "app/i18n";
import { httpClient } from "app/services";
import { createMemoryHistory } from "history";
import React from "react";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import { StubStorage } from "tests/mocks";

import envFixture from "../tests/fixtures/env/data.json";

export const env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage(envFixture) });

const WithProviders: React.FC = ({ children }: { children?: React.ReactNode }) => (
	<I18nextProvider i18n={i18n}>
		<EnvironmentProvider env={env}>{children}</EnvironmentProvider>
	</I18nextProvider>
);

const customRender = (component: React.ReactElement, options: any = {}) =>
	render(component, { wrapper: WithProviders, ...options });

const renderWithRouter = (
	component: React.ReactElement,
	{ routes = ["/"], history = createMemoryHistory({ initialEntries: routes }), withProviders = true } = {},
) => {
	const RouterWrapper = ({ children }: { children: React.ReactNode }) =>
		withProviders ? (
			<WithProviders>
				<Router history={history}>{children}</Router>
			</WithProviders>
		) : (
			<Router history={history}>{children}</Router>
		);

	return {
		...customRender(component, { wrapper: RouterWrapper }),
		history,
	};
};

export * from "@testing-library/react";

export { customRender as render, renderWithRouter };
