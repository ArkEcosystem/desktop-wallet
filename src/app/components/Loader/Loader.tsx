import { CircularProgressBar } from "app/components/CircularProgressBar";
import { Icon } from "app/components/Icon";
import React from "react";

import { LoaderWrapper, LogoSpinner } from "./Loader.styles";

type LoaderProps = {
	show?: boolean;
};

export const Loader = ({ show }: LoaderProps) => {
	if (!show) return <></>;

	return (
		<LoaderWrapper>
			<LogoSpinner>
				<div className="centered">
					<CircularProgressBar
						className="spin"
						showValue={false}
						size={100}
						strokeWidth={3}
						value={40}
						progressColor="var(--theme-color-neutral-800)"
						strokeColor="var(--theme-color-neutral-300)"
					/>
				</div>

				<div className="centered">
					<CircularProgressBar
						className="spin left"
						showValue={false}
						size={75}
						strokeWidth={3}
						value={30}
						progressColor="var(--theme-color-neutral-800)"
						strokeColor="var(--theme-color-neutral-300)"
					/>
				</div>

				<div className="centered">
					<Icon name="LoaderLogo" width={50} height={60} />
				</div>
			</LogoSpinner>
		</LoaderWrapper>
	);
};

Loader.defaultProps = {
	show: true,
};
