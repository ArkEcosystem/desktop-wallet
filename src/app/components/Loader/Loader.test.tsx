import React from "react";
import { render } from "testing-library";

import { Loader } from "./Loader";

describe("Loader", () => {
	it("should render", () => {
		const { container, asFragment } = render(<Loader />);

		expect(container).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render closed", () => {
		const { container, asFragment } = render(<Loader show={false} />);

		expect(container).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});
});
