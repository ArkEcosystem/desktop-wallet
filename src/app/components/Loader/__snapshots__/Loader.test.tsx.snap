// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render 1`] = `
<DocumentFragment>
  <div
    class="Loaderstyles__LoaderWrapper-sc-5aw57o-1 iOGWME"
  >
    <div
      class="Loaderstyles__LogoSpinner-sc-5aw57o-0 jcyvnc"
    >
      <div
        class="centered"
      >
        <svg
          class="spin"
          height="100"
          viewBox="0 0 100 100"
          width="100"
        >
          <circle
            cx="50"
            cy="50"
            fill="transparent"
            r="48.5"
            stroke="var(--theme-color-secondary-300)"
            stroke-width="3"
          />
          <circle
            cx="50"
            cy="50"
            fill="transparent"
            r="48.5"
            stroke="var(--theme-color-secondary-800)"
            stroke-dasharray="304.7344873982099"
            stroke-dashoffset="182.84069243892594"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
            transform="rotate(-90 50 50)"
          />
        </svg>
      </div>
      <div
        class="centered"
      >
        <svg
          class="spin left"
          height="75"
          viewBox="0 0 75 75"
          width="75"
        >
          <circle
            cx="37.5"
            cy="37.5"
            fill="transparent"
            r="36"
            stroke="var(--theme-color-secondary-300)"
            stroke-width="3"
          />
          <circle
            cx="37.5"
            cy="37.5"
            fill="transparent"
            r="36"
            stroke="var(--theme-color-secondary-800)"
            stroke-dasharray="226.1946710584651"
            stroke-dashoffset="158.33626974092556"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
            transform="rotate(-90 37.5 37.5)"
          />
        </svg>
      </div>
      <div
        class="centered"
      >
        <div
          class="sc-bdfBwQ bfbqgS"
          height="60"
          width="50"
        >
          <svg>
            loader-logo.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Loader should render closed 1`] = `<DocumentFragment />`;
